src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,969 @@
+            logger.warning("We found that you are on an older agent version that is not supported.\n Please visit this link to know the agent version support policy 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/agent-upgrade#version-support-policy'.\n")
+            diagnoser_output.append("We found that you are on an older agent version that is not supported.\n Please visit this link to know the agent version support policy 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/agent-upgrade#version-support-policy'.\n")
+            return consts.Diagnostic_Check_Failed
+
+        return consts.Diagnostic_Check_Passed
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to check the azure arc agents version in the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Agent_Version_Check_Fault_Type, summary="Error occured while performing the agent version check")
+        diagnoser_output.append("An exception has occured while trying to check the azure arc agents version in the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete
+
+
+def check_diagnoser_container(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp, storage_space_available, absolute_path, probable_sufficient_resource_for_agents, helm_client_location, kubectl_client_location, release_namespace, security_policy_present):
+
+    global diagnoser_output
+    try:
+
+        if probable_sufficient_resource_for_agents is False:
+            logger.warning("Unable to execute the diagnoser job in the cluster. It may be caused due to insufficient resource availability on the cluster.\n")
+            diagnoser_output.append("Unable to execute the diagnoser job in the cluster. It may be caused due to insufficient resource availability on the cluster.\n")
+            return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+        # Setting DNS and Outbound Check as working
+        dns_check = "Starting"
+        outbound_connectivity_check = "Starting"
+
+        # Executing the Diagnoser job and fetching the logs obtained
+        diagnoser_container_log = executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp, storage_space_available, absolute_path, helm_client_location, kubectl_client_location, release_namespace, security_policy_present)
+
+        # If diagnoser_container_log is not empty then only we will check for the results
+        if(diagnoser_container_log is not None):
+            diagnoser_container_log_list = diagnoser_container_log.split("\n")
+            diagnoser_container_log_list.pop(-1)
+            dns_check_log = ""
+            temp = 1
+            # For fetching the DNS Logs
```

Comment:
it's not just DNS logs right? should you modify the comment to say fetching diagnoser logs

Reply:
I am fetching the DNS logs only in this part. Because the output of nslookup has multiple lines

Reply:
let's syncup offline about this code